components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -142,20 +123,12 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 			projectsDataByte:    projectsDataByte,
 		}
 
-		if len(node.GetSourceId()) == 0 || len(node.GetSourceAccountId()) == 0 || len(node.GetSourceRegion()) == 0 {
+		if !hasCloudInformation(node) {
 			err = tx.upsertByID(node, nodeDetails)
 		} else {
 			node.Uuid, err = tx.upsertByCloudDetails(node, nodeDetails)
 			if err != nil {
-				if pgerr, ok := err.(*pq.Error); ok {
-					if pgerr.Code == pq.ErrorCode("23505") {
-						logrus.Debugf("got duplicate uuid error when attempting upsert. updating by id")
-						// start a new transaction b/c we can't continue the transaction after a failed one
-						return Transact(db, func(tx *DBTrans) error {
```

Comment:
This is the troublesome line that we are trying to eliminate. We do not want a transact inside a transact.